internal/proposals/logs/improvement-of-log-messages-usability.md

```diff
@@ -0,0 +1,92 @@
+
+Users, developers, and maintainers want to have an easy way to track events happening in Kyma, problems that occur, and debug faulty components. This will make understanding of Kyma's ecosystem easier and more transparent
+
+## Solution regarding the log messages
+
+Every component should produce log entries in order to make the problem diagnosis process quick and effective
+
+Key assumptions for the log messages:
+
+- Transparency - every entry must be as short as possible but containing all essential information about the process
+- Tracing - entries sequence must be connected so there is an easy way to filter logs concerning a specific operation
+- GDPR-compliance - entries must not contain any sensitive data (password, personal data in the meaning of GDPR)
+- No "responsibility overlapping" - Because there are other tools to help in diagnosis, logs, in general, should not contain the information which can be taken from that tools (i.e there's tracing which can be used for timings)
+
+Additional assumptions:
+
+- If there is one quick solution for an occurring issue, add it to the message on the DEBUG level so the operator could use it while debugging
+- If operation will be retried, say when
+- JSON format should be enabled only on environments where the monitoring is enabled. Text format is more readable for debugging purposes so make it configurable
+- In case of debug level add the error stack tracing
+- Don't put redundant data into the logs. If something failed 10 times, log that it failed 10 times instead of printing the same error 10 times
+
+## Log structure
+
+To unify the logs, therefore to make debugging process much easier and to make logs parsing also super easy, I'd like to propose a single log format so every service, job and all the components could follow:
+
+- timestamp - RFC3339 Date and time with timezone. For example: "2012-12-12T07:20:50.52Z"
+- level - logging level. For example: "ERROR"
+- message - information with the presented format (notice that there no additional data that could be duplicated in the context structure):
+    - error and fatal message: Past tense started with for example "Failed to...", after that the error wrapped with some meaningful context but without additional "failed to" or "error occurred". For example: "Failed to provision runtime: while fetching release: while validating relese: release does not contain installer yaml"
+    - info message: Present Continuous tense for the things that are about to be done, for example "Starting processing..." or past tense for the things that are finished, like "Finished successfully!"
+    - notice and warning message: A short explanation on what happened and what this can cause. For example: "Tiller configuration not found in the release artifacts. Proceeding to the Helm 3 installation..." or "Connection is not yet established. Retrying in 5 minutes..."
```

Comment:
```suggestion
    - notice and warning message: a short explanation of what happened and what this can cause. For example: `Tiller configuration not found in the release artifacts. Proceeding to the Helm 3 installation...` or `Connection is not yet established. Retrying in 5 minutes...`.
```